public/app/features/dashboard/dashboard_ctrl.ts

```diff
@@ -29,11 +30,15 @@ export class DashboardCtrl {
       var resizeEventTimeout;
 
       $scope.setupDashboard = function(data) {
-        var dashboard = dashboardSrv.create(data.dashboard, data.meta);
-        dashboardSrv.setCurrent(dashboard);
-
-        // init services
-        timeSrv.init(dashboard);
+        try {
+          var dashboard = dashboardSrv.create(data.dashboard, data.meta);
+          dashboardSrv.setCurrent(dashboard);
+
+          // init services
+          timeSrv.init(dashboard);
+        } catch (err) {
+          $scope.appEvent("alert-error", ['Dashboard init failed', err.message]);
```

Comment:
But, the exception handler you mentioned is Promise's one.
I don't know well about it, is it possible to combine these exception handler?

https://github.com/mtanda/grafana/blob/44b2a292eb1e467b07b2941246ab8e0e153dffd3/public/app/features/dashboard/dashboard_ctrl.ts#L45